src/auth/Crypto.cc

```diff
@@ -420,6 +420,8 @@ int CryptoKey::_set_secret(int t, const bufferptr& s)
     if (error.length()) {
       return -EIO;
     }
+  } else {
+      return -EOPNOTSUPP;
```

Comment:
Could you add a test that reproduce the problem and shows this fixes it ? If you're not sure where such a test should be added, let me know.